source/lac/matrix_scaling.cc

```diff
@@ -0,0 +1,440 @@
+          column_scaling[col] /= std::sqrt(col_norms[col]);
+        }
+    }
+}
+
+
+
+template <class Matrix>
+void
+MatrixScaling::sk_scaling(Matrix &matrix, const unsigned int nsteps)
+{
+  using Number = typename Matrix::value_type;
+
+  Vector<Number> row_norms(matrix.m());
+  Vector<Number> col_norms(matrix.n());
+
+  switch (control.sinkhorn_knopp_parameters.norm_type)
+    {
+      case MatrixScaling::AdditionalData::SKParameters::NormType::l1:
+        {
+          // Row_norms_0 to start the procedure
+          row_norms = 0;
+          for (unsigned int row = 0; row < matrix.m(); ++row)
+            for (auto it = matrix.begin(row); it != matrix.end(row); ++it)
+              row_norms[row] += std::abs(it->value());
+
+          for (unsigned int i = 0; i < nsteps; i++)
+            {
+              // Row step
+              col_norms = 0;
+              for (unsigned int row = 0; row < matrix.m(); ++row)
+                {
+                  for (auto it = matrix.begin(row); it != matrix.end(row); ++it)
+                    {
+                      // it->value() /= row_norms[row];
+                      matrix.set(row,
+                                 it->column(),
+                                 it->value() / row_norms[row]);
+                      col_norms[it->column()] += std::abs(it->value());
+                    }
+                  row_scaling[row] /= row_norms[row];
+                }
+              // Here exit condition only on columns, rows are already
+              // normalized
+              col_norms.add(-1.0);
+              if (col_norms.l1_norm() < control.scaling_tolerance)
+                {
+                  converged = true;
+                  break;
+                }
+              col_norms.add(1.0);
+
+              // Column step
+              row_norms = 0;
+              for (unsigned int row = 0; row < matrix.m(); ++row)
+                {
+                  for (auto it = matrix.begin(row); it != matrix.end(row); ++it)
+                    {
+                      // it->value() /= col_norms[it->column()];
+                      matrix.set(row,
+                                 it->column(),
+                                 it->value() / col_norms[it->column()]);
+                      row_norms[row] += std::abs(it->value());
+                    }
+                }
+              for (unsigned int col = 0; col < matrix.n(); ++col)
+                {
+                  column_scaling[col] /= col_norms[col];
+                }
+              // Here exit condition only on rows, columns are already
+              // normalized
+              row_norms.add(-1.0);
+              if (row_norms.l1_norm() < control.scaling_tolerance)
+                {
+                  converged = true;
+                  break;
+                }
+              row_norms.add(1.0);
+            }
+        }
+        break;
+      case MatrixScaling::AdditionalData::SKParameters::NormType::l_infinity:
+        {
```

Comment:
It would be easier to read with the `break` statement inside the curly braces:
```suggestion
          break;
        }

            case MatrixScaling::AdditionalData::SKParameters::NormType::l_infinity:
        {
```
I also gave it an extra line to make it easier to read.